### PR TITLE
Add profiles count report

### PR DESCRIPTION
**Pull request by seanwiseman**
Enables the following:

- `manage.py get_profiles_count` command that will go get the latest profiles count total and store the time stamped value.
- `reports/profile-counts` endpoint that currently returns a CSV formatted response of all profile count data.

### CODE REVIEW

File: src/observer/tests/test_views.py [hunk 1]
```diff
@@ -112,7 +112,12 @@ def test_report_format_hint(self):
                 # test the content
                 if format == reports.CSV:
                     # it's a list of strings with commas in it.
-                    list(resp.streaming_content)[0].decode('utf8').split(',')[0]
+                    try:
+                        list(resp.streaming_content)[0].decode('utf8').split(',')[0]
+                    except AttributeError:
+                        # AttributeError: 'HttpResponse' object has no attribute 'streaming_content'
```
Comment (seanwiseman) [inline src/observer/tests/test_views.py]: @lsh-0 could do with some assistance with this.

File: src/observer/management/commands/get_profiles_count.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import sys, json
```
Comment (giorgiosironi) [inline src/observer/management/commands/get_profiles_count.py]: `get` can be counter-intuitive for a command that does not just read the API but also stores values in the database

File: src/observer/models.py [hunk 3]
```diff
@@ -190,3 +190,16 @@ def __str__(self):
 
     def __repr__(self):
         return '<ArticleJSON "%s">' % self
+
+class ProfileCount(models.Model):
+    total = PositiveIntegerField(null=False, blank=False)
+    timestamp = DateTimeField(auto_now_add=True)
```
Comment (giorgiosironi) [inline src/observer/models.py]: does this mean the time is generated on the Python object or in the database? The former usually gives better control and reproducibility, e.g. we are able to insert data retroactively if needed

File: src/observer/reports.py [hunk 4]
```diff
@@ -105,6 +105,23 @@ def published_research_article_index():
         .order_by('msid') \
         .values_list('msid', 'datetime_poa_published', 'datetime_vor_published')
 
+@report({
+    'title': "profile counts",
+    'description': "Daily record of the total number of profiles",
+    'serialisations': [CSV],
+    'per_page': NO_PAGINATION,
```
Comment (giorgiosironi) [inline src/observer/reports.py]: pagination can probably be introduced later (if needed)

**Comment by lsh-0** [comment]
a couple of thoughts: 

* nix the new management command and make the existing load_from_api command smarter. management commands tend to spread and repeat boilerplate code.
* just capturing a count? you have a list of profiles with unique IDs. I would capture and store each profile with a timestamp of 'date created' (first seen) so then over time you can see not just a number growing but who. Granted, the data doesn't look very interesting right now ... but it does have an orcid and we may be able to go slurp the rest of the data later.

I'll address these two points today and take a look at those tests. 

This branch could take advantage of my work in the press-packages branch

**Comment by lsh-0** [comment]
ok, I've integrated your changes into the press-packages branch I was working on and changed how Profiles are captured: https://github.com/elifesciences/observer/pull/37

I got rid of the management command and integrated the actions (scraping) into the load_from_api management command. as @giorgiosironi mentioned, we have a daily metrics scrape as well, so I might just create a `daily.sh` script with the correct incantations for cron to run. 

I have two of my own tests failing after the merge and need to address some of your tests as well so I'm working on that now.

The press-packages stuff was essentially done sans some test coverage, so hopefully the whole shebang should be ready for review by the time you guys come online.

**Comment by lsh-0** [comment]
PR for daily cron job: https://github.com/elifesciences/observer-formula/pull/6

**Comment by seanwiseman** [comment]
Closing as work has been pulled into [#37](https://github.com/elifesciences/observer/pull/37)

**Pull request closed without merging**